src/rv/comm/rcssserver/GameState.java

```diff
@@ -48,6 +48,34 @@
         void gsServerMessageProcessed(GameState gs);
     }
 
+    public enum FoulType {
+        CROWDING(0, "crowding"), TOUCHING(1, "touching"), ILLEGALDEFENCE(2, "illegal defence"),
+        ILLEGALATTACK(3, "illegal attack"), INCAPABLE(4, "incapable"),
+        KICKOFF(5, "illegal kickoff"), CHARGING(6, "charging");
+
+        private int    idx;
```

Comment:
Wouldn't that make the `index` var `Foul` has redundant (since the index is already contained in the `type`)?